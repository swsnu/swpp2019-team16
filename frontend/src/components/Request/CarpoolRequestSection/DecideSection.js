import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import Button from '../../common/Button';
import { Typography } from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    marginBottom: '25px',
  },
  content: {},
});

DecideSection.propTypes = {
  onClick: PropTypes.func.isRequired,
};

function DecideSection({ onClick }) {
  const styles = useStyles();
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h2'}>Are you ready to call Ya-Ta?</Typography>
      </div>
      <div className={styles.content}>
        <Button
          id="request-submit-button"
          variant="contained"
          color="secondary"
          onClick={onClick}
          children="Send Request"
        />
      </div>
    </div>
  );
}

export default DecideSection;
