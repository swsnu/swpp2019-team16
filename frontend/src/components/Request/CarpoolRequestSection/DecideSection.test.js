import React from 'react';
import { render } from '@testing-library/react';
import DecideSection from './DecideSection';

describe('<DecideSection />', () => {
  it('should match with snapshot - default', async () => {
    const { container } = render(<DecideSection onClick={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
