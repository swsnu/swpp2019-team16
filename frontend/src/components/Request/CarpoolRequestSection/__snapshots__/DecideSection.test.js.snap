// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DecideSection /> should match with snapshot - default 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Are you ready to call Ya-Ta?
      </h2>
    </div>
    <div
      class="makeStyles-content-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-34 makeStyles-root-37 MuiButton-contained makeStyles-contained-35 makeStyles-contained-38 MuiButton-contained makeStyles-contained-35 makeStyles-contained-38"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Send Request
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
